src/sentry/scim/endpoints/teams.py

```diff
@@ -0,0 +1,213 @@
+
+        if "members" in request.GET.get("excludedAttributes", []):
+            exclude_members = True
+        else:
+            exclude_members = False
+        queryset = Team.objects.filter(
+            organization=organization, status=TeamStatus.VISIBLE
+        ).order_by("slug")
+
+        if filter_val:
+            queryset = queryset.filter(slug=slugify(filter_val))
+
+        def data_fn(offset, limit):
+            return list(queryset[offset : offset + limit])
+
+        def on_results(results):
+            results = serialize(
+                results, None, TeamSCIMSerializer(), exclude_members=exclude_members
+            )
+            return self.list_api_format(request, queryset, results)
+
+        return self.paginate(
+            request=request,
+            on_results=on_results,
+            paginator=GenericOffsetPaginator(data_fn=data_fn),
+            default_per_page=int(request.GET.get("count", 100)),
+            queryset=queryset,
+            cursor_cls=SCIMCursor,
+        )
+
+    def post(self, request, organization):
+        # shim displayName from SCIM api to "slug" in order to work with
+        # our regular team index POST
+        request.data.update({"slug": slugify(request.data["displayName"])})
+        return super().post(request, organization)
+
+
+class OrganizationSCIMTeamDetails(SCIMEndpoint, TeamDetailsEndpoint):
+    permission_classes = (OrganizationSCIMTeamPermission,)
+
+    def convert_args(self, request, organization_slug, team_id, *args, **kwargs):
+        args, kwargs = super().convert_args(request, organization_slug)
+        try:
+            kwargs["team"] = self._get_team(kwargs["organization"], team_id)
+        except Team.DoesNotExist:
+            raise ResourceDoesNotExist
+        return (args, kwargs)
+
+    def _get_team(self, organization, team_id):
+        team = (
+            Team.objects.filter(organization=organization, id=team_id)
+            .select_related("organization")
+            .get()
+        )
+        if team.status != TeamStatus.VISIBLE:
+            raise Team.DoesNotExist
+        return team
+
+    def get(self, request, organization, team):
+        context = serialize(team, serializer=TeamSCIMSerializer())
+        return Response(context)
+
+    def _add_members_operation(self, request, operation, team):
+        for member in operation["value"]:
+            member = OrganizationMember.objects.get(
+                organization=team.organization, id=member["value"]
+            )
+            with transaction.atomic():
+                omt = OrganizationMemberTeam.objects.create(team=team, organizationmember=member)
+                self.create_audit_entry(
+                    request=request,
+                    organization=team.organization,
+                    target_object=omt.id,
+                    target_user=member.user,
+                    event=AuditLogEntryEvent.MEMBER_JOIN_TEAM,
```

Comment:
Would it make sense to differentiate between SCIM API vs other endpoints for the audit log?

Reply:
<img width="850" alt="Screen Shot 2021-06-15 at 10 59 55 AM" src="https://user-images.githubusercontent.com/1976777/122101171-da9fef80-cdc8-11eb-817c-818829efd8e4.png">
so the audit log will display the name of the internal integration and we can query based on that -- should be enough to delineate